src/john.c

```diff
@@ -1094,7 +1094,10 @@ static void john_load(void)
 			options.loader.flags |= DB_CRACKED;
 			ldr_init_database(&database, &options.loader);
 
-			if (!options.loader.showformats) {
+			if (options.loader.showformats) {
+				if (!options.loader.showformats_old)
+					fputs("[", stdout);
```

Comment:
FWIW, this could be `putchar`.

Reply:
True. I got the idea to look for similar cases and the only other one is a single place in uaf_hash.c. I could fix them with almost no effort if you like but it doesn't matter at all, right?

Reply:
This is unimportant indeed. Up to you. We have plenty of more important code cleanups to make anyway.